### DISCUSSION TITLE
Access log to output plugin variables

### DISCUSSION THREAD

**Discussion by MUI-Pop** [Kong/kong#13311] [Help] [answered]
I have few metadata as variables in plugin that need to be outputted in kong's access log. What is the best way to share the variable between plugin and kong access log ?

**Comment by hanshuebner** [comment, marked as answer]
Hello @MUI-Pop,

it is difficult to say what the best way is as that depends on your environment and deployment model.  If you need to change the format of the `access_log` file which is written by nginx, then you'll need to work with a [custom nginx configuration template](https://docs.konghq.com/gateway/latest/production/logging/customize-gateway-logs/), define the new variables and the access log format to use those variables, and then set the variables from your plugin in `ngx.ctx`.  This approach is a little fiddly, however, in particular as you'll need to make sure that your custom templates keeps working across upgrades. 

If you have a little more freedom with respect to how your access log is written, you could use a custom plugin to write the access log in the format that you require.  That plugin could then access `ngx.ctx` to retrieve whatever information has been left there by your other plugin, or your other plugin could use [`kong.log.set_serialize_value()`](https://docs.konghq.com/gateway/latest/plugin-development/pdk/kong.log/#konglogset_serialize_valuekey-value-options) to put information into the `log.serialize()` return value.

Does this help?

**Reply by chobits** [reply]
hi @MUI-Pop  I have marked this as answer, could you check it?

**Reply by MUI-Pop** [reply]
Thank you @hanshuebner. We are using the `access_log` nginx descriptor, in this case using `ngx.ctx` for passing data seems the way.

**Comment by MUI-Pop** [comment]
Commented too early, looks like `ngx.ctx`cannot be interpolated in the access log, hence I declared a `ngx.var` and referred it in access log.
